app/javascript/crayons/MentionAutocompleteTextArea/MentionAutocompleteTextArea.jsx

```diff
@@ -54,284 +73,311 @@ const replaceTextArea = ({
-          value={textContent}
-          data-mention-autocomplete-active="true"
-          as="textarea"
-          autocomplete={false}
-          onChange={handleTextInputChange}
-        />
-
-        <textarea
-          data-mention-autocomplete-active="true"
-          ref={plainTextAreaRef}
-          onChange={handleTextInputChange}
-          value={textContent}
-        />
-
-        {searchTerm && (
-          <ComboboxPopover
-            ref={popoverRef}
-            className="crayons-autocomplete__popover absolute"
-            id="mention-autocomplete-popover"
-            style={{
-              top: `calc(${dropdownPositionPoints.y}px + 1.5rem)`,
-              left: `${dropdownPositionPoints.x}px`,
+    }, [replaceElement]);
+
+    return (
+      <Fragment>
+        <div aria-live="polite" class="screen-reader-only">
+          {ariaHelperText}
+        </div>
+
+        <Combobox
+          ref={containerRef}
+          id="combobox-container"
+          onSelect={handleSelect}
+          className={`crayons-autocomplete${autoResize ? ' h-100' : ''}`}
+        >
+          <ComboboxInput
+            {...autocompleteInputProps}
+            aria-label="Mention user"
+            ref={comboboxRef}
+            value={textContent}
+            data-mention-autocomplete-active="true"
+            as="textarea"
+            autocomplete={false}
+            onChange={(e) => {
+              onChange?.(e);
+              handleTextInputChange(e);
             }}
-          >
-            {users.length > 0 ? (
-              <ComboboxList>
-                {users.map((user) => (
-                  <ComboboxOption
-                    value={user.username}
-                    className="crayons-autocomplete__option flex items-center"
-                  >
-                    <UserListItemContent user={user} />
-                  </ComboboxOption>
-                ))}
-              </ComboboxList>
-            ) : (
-              <span className="crayons-autocomplete__empty">
-                {searchTerm.length >= MIN_SEARCH_CHARACTERS
-                  ? 'No results found'
-                  : 'Type to search for a user'}
-              </span>
-            )}
-          </ComboboxPopover>
-        )}
-      </Combobox>
-    </Fragment>
-  );
-};
+          />
+
+          <textarea
+            {...autocompleteInputProps}
+            id={inputId}
+            data-mention-autocomplete-active="true"
+            ref={mergeInputRefs([plainTextAreaRef, forwardedRef])}
+            onChange={(e) => {
+              onChange?.(e);
+              handleTextInputChange(e);
+            }}
+            value={textContent}
+          />
+
+          {searchTerm && (
+            <ComboboxPopover
+              ref={popoverRef}
+              className="crayons-autocomplete__popover absolute"
+              id="mention-autocomplete-popover"
+              style={{
+                top: `calc(${dropdownPositionPoints.y}px + 1.5rem)`,
```

Comment:
Would it make sense to pull out `1.5rem` into a constant here? From my past experiences, pulling out CSS/class-related styles into a const or some easy-to-change variable makes it much easier to find and debug and tweak in the future! I've been lost in a JS file more times than I'd care to admit, trying to figure out where certain styles are coming from, and I could see this one easily being missed/buried in the JSX itself.

Reply:
done!